src/it/scala/net/snowflake/spark/snowflake/SnowflakeTelemetrySuite.scala

```diff
@@ -0,0 +1,95 @@
+package net.snowflake.spark.snowflake
+
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.ObjectMapper
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.ObjectNode
+import net.snowflake.client.jdbc.telemetry.Telemetry
+import net.snowflake.spark.snowflake.Utils.SNOWFLAKE_SOURCE_NAME
+import org.apache.spark.SparkEnv
+import org.apache.spark.sql.{Row, SaveMode}
+import org.apache.spark.sql.types._
+
+import scala.collection.mutable
+import scala.util.Random
+
+private[snowflake] class FakeTelemetryMessageSender(buffer: mutable.ArrayBuffer[ObjectNode])
```

Comment:
With the fake `TelemetryMessageSender`, the telemetry messages are appended into the ArrayBuffer instead of sending to snowflake, and the test can check the message to be sent.

Reply:
how about `MockTelemetryMessageSender`?

Reply:
Will update to use `MockTelemetryMessageSender` in next PR update.